engine/src/main/scala/cromwell/engine/backend/CanUseGcsFilesystem.scala

```diff
@@ -0,0 +1,44 @@
+package cromwell.engine.backend
+
+import java.nio.file.FileSystem
+
+import cromwell.core.WorkflowOptions
+import cromwell.filesystems.gcs.{GcsFileSystemProvider, GoogleAuthMode, GoogleConfiguration}
+import lenthall.config.ScalaConfig._
+import EnhancedWorkflowOptions._
+
+import scalaz.NonEmptyList
+
+trait CanUseGcsFilesystem { self: Backend =>
+
+  protected def gcsFilesystem(options: WorkflowOptions): Option[FileSystem] = {
+
+    def gcsFilesystemMustHaveAuth: Option[String] = {
+      backendConfig.getConfigOption("filesystems.gcs") map {
+        _.getStringOr("auth", throw new RuntimeException(s"No auth in filesystems.gcs for backend '$name'.")) }
+    }
+
+    def gcsFilesystemAuthMustBeLegit(authName: String): GoogleAuthMode = {
+      def prependName(errs: NonEmptyList[String]): String = s"For backend '$name': " + errs.list.mkString("\n")
```

Comment:
Lenthall doesn't seem to already have something that exactly fits this, but yeah:

```scala
class ErrorMessageAggregatedException(override val exceptionContext: String,
                                      override val errorMessages: Traversable[String]) extends MessageAggregation

GoogleConfiguration.Instance.auth(authName) valueOr { errs => throw new ErrorMessageAggregatedException(s"For backend '$name': ", errs.list) }
```